### PR TITLE
chore(docker-desktop-extensions): add methods to call REST API

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
part of #2397
here it allows to call REST API methods

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

#2397 

### How to test this PR?

unit tests added

### CODE REVIEW

File: packages/main/src/plugin/docker-extension/docker-desktop-installation.ts [hunk 1]
```diff
@@ -310,5 +313,78 @@ export class DockerDesktopInstallation {
...
+        // provides the body as is
+        return response.body;
+      }
+    } catch (requestErr) {
+      if (
+        requestErr instanceof RequestError &&
+        (requestErr.response?.statusCode === 401 || requestErr.response?.statusCode === 403)
+      ) {
+        throw Error('Unable to get access');
+      } else if (requestErr instanceof Error) {
```
Comment (lstocchi) [inline packages/main/src/plugin/docker-extension/docker-desktop-installation.ts:385]: Doesn't it miss a generic else branch? May it have a different type than Error and RequestError?

**Review by lstocchi — APPROVED** [review]
LGTM, just a note below

File: packages/main/src/plugin/docker-extension/docker-desktop-installation.ts (hunk 1 above)
Comment (benoitf) [reply packages/main/src/plugin/docker-extension/docker-desktop-installation.ts:385]: good catch, I have added else branch in a follow-up commit + unit test

Comment (benoitf) [reply packages/main/src/plugin/docker-extension/docker-desktop-installation.ts:385]: in fact we have TypeScript limitation as got only send RequestError anyway...

**Review by lstocchi — APPROVED** [review]

**Pull request merged**